### ISSUE TITLE
wrong location for os provider in Windows

### ISSUE THREAD

**Issue by atomic111** [mondoohq/cnquery#2460]
if i run the cnspec serve in windows server the path of the os provider is in `C:\Windows\System32\config\systemprofile\.config\mondoo\providers\os`

my expectation is that the provider is locate in `C:\ProgramData\Mondoo\providers`

cnspec 9.5.1 (e8298bf, 2023-11-02T10:07:22Z)

**Comment by chris-rock** [comment]
Right now providers are installed in the user profile that runs the binary. In this case the service runs under system user hence the local providers in that location right now. We should prefer the `C:\ProgramData\Mondoo\providers` if we have write permissions to that directory.

**Comment by vjeffrey** [comment]
is the todo here to add ` C:\ProgramData\` as a home folder option for windows if we have perms to write to it?

**Comment by atomic111** [comment]
@vjeffrey Yes, i think so. If the mondoo service runs in Windows, it is executed as the NTAuthority user. This user has the highest permissions within Windows.

**Comment by chris-rock** [comment]
Using  C:\Windows\System32\config\systemprofile\.config\mondoo\providers\os for the os provider is sub-optimal. What we want:

- use home directory for all users except for system user
- system user uses global `C:\ProgramData\Mondoo\providers` as default

For the migration we should remove the `C:\Windows\System32\config\systemprofile\.config\mondoo` directory. We should not fail the migration when the directory does not exist or when we cannot remove the directory (e.g. via missing permissions).

**Comment by tas50** [comment]
Reviewed as part of a full backlog triage — **confirmed still open** in the current codebase.

providers/providers.go:52-58 init() sets DefaultPath=SystemPath then overrides to HomePath when 'os.Geteuid() != 0' with no Windows guard. On Windows os.Geteuid() returns -1, so DefaultPath always becomes HomePath (~/.config/mondoo => systemprofile\.config\mondoo for a service account). cli/config/path.go:82-103 systemPath() correctly returns C:\ProgramData\Mondoo\ for windows, but the init never picks SystemPath on Windows. No runtime.GOOS=="windows" handling exists at providers.go:55 (only later at lines 806/854/1009).

**Suggested fix approach:** In providers/providers.go init() guard the Geteuid()-based HomePath override: on Windows (runtime.GOOS=="windows") prefer SystemPath (C:\ProgramData\Mondoo from cli/config systemPath), since Geteuid() is meaningless there. Determine elevation via a Windows-appropriate check (or just default to SystemPath for services) instead of os.Geteuid()!=0.
